collections/ansible_collections/cloudkit/service/roles/hostpool_infra/tasks/create_hostpool_infra.yaml

```diff
@@ -0,0 +1,107 @@
+- name: Show host requests
+  ansible.builtin.debug:
+    var: hostpool_infra_host_requests
+
+- name: Create hostpool network
+  ansible.builtin.include_role:
+    name: massopencloud.esi.network
+  vars:
+    network_suffix: "{{ hostpool_infra_name }}"
+    network_state: present
+    network_properties:
+      instance: "{{ hostpool_infra_namespace }}"
+      hostpool: "{{ hostpool_infra_name }}"
+      purpose: "o-sac"
+
+- name: Set hostpool_infra_network_name
+  ansible.builtin.set_fact:
+    hostpool_infra_network_name: "network-{{ hostpool_infra_name }}"
+
+# Handle scale down **before** scale up
+
+- name: List host classes currently in use by hostpool
+  ansible.builtin.include_role:
+    name: massopencloud.esi.host
+    tasks_from: list_host_classes_for_hostpool
+  vars:
+    host_pool_name: "{{ hostpool_infra_name }}"
+
+- name: Set host classes included in the request
+  ansible.builtin.set_fact:
+    requested_host_classes: >
+      {{ hostpool_infra_host_requests | map(attribute='hostClass') }}
+
+- name: Determine which host classes are no longer needed
+  ansible.builtin.set_fact:
+    removed_host_classes: "{{ removed_host_classes | default([]) + [item] }}"
+  with_items: "{{ host_class_list }}"
+  when: item not in requested_host_classes
```

Comment:
There is a `difference` filter in Jinja:

```
- hosts: localhost
  gather_facts: false
  vars:
    host_class_list:
      - classA
      - classB
      - classC
    requested_host_classes:
      - classA
      - classD

  tasks:
    - ansible.builtin.set_fact:
        removed_host_classes: >-
          {{ host_class_list | difference(requested_host_classes) }}

    - name: Display host classes to be removed
      ansible.builtin.debug:
        msg: "The following host classes are no longer requested: {{ removed_host_classes | default([]) }}"
```

Reply:
Fixed!